src/output-json.c

```diff
@@ -1179,27 +1178,51 @@ OutputInitResult OutputJsonInitCtx(ConfNode *conf)
             }
 
             if (SCConfLogOpenRedis(redis_node, json_ctx->file_ctx) < 0) {
+                SCFree(json_ctx->file_ctx->sensor_name);
                 LogFileFreeCtx(json_ctx->file_ctx);
                 SCFree(json_ctx);
                 SCFree(output_ctx);
                 return result;
             }
         }
 #endif
+#ifdef HAVE_PLUGINS
         else if (json_ctx->json_out == LOGFILE_TYPE_PLUGIN) {
-            ConfNode *plugin_conf = ConfNodeLookupChild(conf,
-                json_ctx->plugin->name);
-            void *plugin_data = NULL;
-            if (json_ctx->plugin->Open(plugin_conf, &plugin_data) < 0) {
+            void *init_data = NULL;
+            const char *append = ConfNodeLookupChildValue(conf, "append");
+            if (append == NULL) {
+                append = DEFAULT_LOG_MODE_APPEND;
+            }
+            if (json_ctx->file_ctx->threaded) {
+                if (!SCLogOpenThreadedFile(
+                            ConfigGetLogDirectory(), append, json_ctx->file_ctx, 1)) {
+                    SCFree(json_ctx);
+                    SCFree(output_ctx);
+                    return result;
+                }
+            }
```

Comment:
Is this needed for plugins?

Reply:
Yes -- this is where Suricata keeps track of the per-thread data blobs. `lf_slots` contains a blob for each thread -- the plugin threadinit returns the data blob tracked by Suricata.

Reply:
Ok.  I commented it out and things seem to work OK.  I think it needs a comment, cause the `append`, which is not really needed in the plugin block and naming make it sound out of place in this section of code.

Reply:
> Ok. I commented it out and things seem to work OK. I think it needs a comment, cause the `append`, which is not really needed in the plugin block and naming make it sound out of place in this section of code.

My error.. Must have had threading off when I gave this a try.